Intersect.Client.Core/Utilities/TextColorParser.cs

```diff
@@ -0,0 +1,69 @@
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Intersect.Client.Utilities
+{
+    public static class TextColorParser
+    {
+        private static readonly Dictionary<string, Func<Color>> KnownColors = InitializeColorDictionary();
+        private static readonly Regex ColorTagRegex = new(@"\\c{([^}]*)}", RegexOptions.Compiled);
+        private static readonly string SplitPattern = @"(\\c{[^}]*})";
+
+        private static Dictionary<string, Func<Color>> InitializeColorDictionary()
+        {
+            return typeof(Color)
+                .GetProperties(BindingFlags.Public | BindingFlags.Static)
+                .Where(p => p.PropertyType == typeof(Color))
+                .ToDictionary(
+                    p => p.Name.ToLowerInvariant(),
+                    p => (Func<Color>)(() => p.GetMethod?.Invoke(null, null) as Color ?? Color.White)
+                );
+        }
+
+        public static List<ColorizedText> Parse(string text, Color defaultColor)
+        {
+            var output = new List<ColorizedText>();
```

Comment:
Initialize this after `var segments` and pass `segments.Length` in as a parameter -- we know exactly how many segments are going to be in this list, no reason to not start the capacity at that value and avoid re-allocation